elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchIndexType.scala

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.elasticsearch
+
+case class ElasticsearchIndexType(indexName: String, typeName: Option[String], apiVersion: ApiVersion) {
```

Comment:
Should this become the type to set index and type instead?

Reply:
I'm not sure that I understand what you're asking here. Could you clarify this for me?

Reply:
The current API takes `indexName: String, typeName: String`. For ES 7 users can ignore `typeName`.
I wonder if this class should become the parameter to pass in these fields in an API version aware way. Roughly `ElasticsearchFlow.create(Index("indexName"), settings)` instead of ``ElasticsearchFlow.create("indexName", null, settings)`.

Reply:
Thanks for clarifying. That's actually something that crossed my mind back when I added the type. Maybe it's a nice idea to add 2 builder methods to the `ElasticsearchIndexType` to make it even more transparent to users what properties are actually required for their ES version:

```
object ElasticsearchIndexType {
  def forElasticsearchV7(indexName: String): ElasticsearchIndexType =
    ElasticsearchIndexType(indexName, None, ApiVersion.V7)

  def forElasticsearchV5(indexName: String, typeName: String): ElasticsearchIndexType =
    ElasticsearchIndexType(indexName, Some(typeName), ApiVersion.V5)
}
```

What do you think?

Reply:
Can we come up with something shorter?

eg `ElasticsearchFlow.create(EsParams.v5("indexName", "typeName"), settings)`

This might help if later versions would allow/require even more things.

Reply:
Sounds good 👍 Will update a.s.a.p.